src/core/balance-sheet.ts

```diff
@@ -79,27 +79,13 @@ export const updateUsersPendingPayment = async ({
     await updatePendingLightningTransactions()
   }
 
-  const users = User.find({ onchain: { $exists: true, $not: { $size: 0 } } }).cursor({
-    batchSize: 100,
-  })
-
-  await runInParallel({
-    iterator: users,
-    logger,
-    workers: 3,
-    processor: async (user, index) => {
-      logger.trace("updating onchain receipt for user %o in worker %d", user._id, index)
-      const result = await Wallets.updateOnChainReceipt(user.id, logger)
-      if (result instanceof Error) {
-        logger.error(
-          { userId: user.id, index },
-          "Could not updateOnChainReceipt from balance-sheet",
-        )
-      }
-    },
-  })
+  const txNumber = await Wallets.updateOnChainReceipt({ logger })
```

Comment:
in this way the lnd load will be reduced at minimum and we will only update users related to the onchain transactions